pkg/source/ttnv2/source.go

```diff
@@ -210,5 +209,8 @@ func (s *Source) RangeDevices(appID string, f func(source.Source, string) error)
 
 // Close implements the Source interface.
 func (s *Source) Close() error {
-	return s.client.Close()
+	if s.client != nil {
```

Comment:
I don't really like the fact that we lazy-initialize `client` with the first application ID that is seen by `Source`. Is there a good reason for that?

Reply:
Lazy initialization is removed as part of #13